hikari/impl/rest.py

```diff
@@ -1481,28 +1481,26 @@ async def delete_messages(
             # (albeit at a cost of maybe a couple-dozen milliseconds per call),
             # I am just gonna invoke these sequentially instead.
             try:
-                if len(pending) == 1:
-                    message = pending[0]
+                if len(chunk) == 1:
+                    message = chunk[0]
                     try:
                         await self.delete_message(channel, message)
-                    except errors.NotFoundError as exc:
+                    except errors.NotFoundError as e:
                         # If the message is not found then this error should be suppressed
                         # to keep consistency with how the bulk delete endpoint functions.
-                        if exc.code != 10008:  # Unknown Message
+                        if e.code != 10008:  # Unknown Message
                             raise
 
                     deleted.append(message)
 
                 else:
                     body = data_binding.JSONObjectBuilder()
-                    chunk = pending[:100]
                     body.put_snowflake_array("messages", chunk)
                     await self._request(route, json=body)
                     deleted += chunk
 
-                pending = pending[100:]
-            except Exception as ex:
-                raise errors.BulkDeleteError(deleted, pending) from ex
+            except Exception as e:
+                raise errors.BulkDeleteError(deleted, await iterator) from e
```

Comment:
Got a little problem with this, as we are effectively consuming the whole iterator. We no longer know it has a finate number of items, so this could potentially go on for ever, which would cause it to never return/raise the exception